src/modules/ekf2/EKF/aid_sources/magnetometer/mag_control.cpp

```diff
@@ -210,11 +210,13 @@ void Ekf::controlMagFusion(const imuSample &imu_sample)
 
 		if (_control_status.flags.mag) {
 
-			if (continuing_conditions_passing && _control_status.flags.yaw_align) {
+			if (continuing_conditions_passing) {
 
 				if (checkHaglYawResetReq() || (wmm_updated && no_ne_aiding_or_not_moving)) {
 					ECL_INFO("reset to %s", AID_SRC_NAME);
-					resetMagStates(_mag_lpf.getState(), _control_status.flags.mag_hdg || _control_status.flags.mag_3D);
+					const bool reset_heading = !_control_status.flags.yaw_align || _control_status.flags.mag_hdg
```

Comment:
What about if `MagFuseType::INIT`?

Reply:
`INIT` will still align because of `!_control_status.flags.yaw_align`. but you're right, it should only align if the checks are passing...

Reply:
It will also reset because of `!_control_status.flags.yaw_align` but then, since mag_hdg never starts, it won't update the heading. Good point, I'll solve this.